vespa/application.py

```diff
@@ -524,31 +538,39 @@ def delete_batch(
                     "Not possible to infer schema name. Specify schema parameter."
                 )
 
+        if not namespace:
+            namespace = schema
+
         if asynchronous:
             coro = self._delete_batch_async(
+                namespace=namespace,
                 schema=schema,
                 batch=batch,
                 connections=connections,
                 total_timeout=total_timeout,
             )
             return self._check_for_running_loop_and_run_coroutine(coro=coro)
         else:
-            return self._delete_batch_sync(schema=schema, batch=batch)
+            return self._delete_batch_sync(namespace=namespace, schema=schema, batch=batch)
 
-    def delete_all_docs(self, content_cluster_name: str, schema: str) -> Response:
+    def delete_all_docs(self, content_cluster_name: str, schema: str,  namespace: str = None) -> Response:
         """
         Delete all documents associated with the schema
 
         :param content_cluster_name: Name of content cluster to GET from, or visit.
         :param schema: The schema that we are deleting data from.
+        :param namespace: The  namespace that we are deleting data from.
         :return: Response of the HTTP DELETE request.
         """
+        if not namespace:
+            namespace = schema
+
         with VespaSync(self) as sync_app:
             return sync_app.delete_all_docs(
-                content_cluster_name=content_cluster_name, schema=schema
+                content_cluster_name=content_cluster_name, namespace=namespace, schema=schema
             )
 
-    def get_data(self, schema: str, data_id: str) -> VespaResponse:
+    def get_data(self, namespace: str, schema: str, data_id: str) -> VespaResponse:
```

Comment:
The changes on this PR should not break the API. Note that in this case `namespace` is a required argument. It should have a default and use schema when not provided.

You can check API breaks by looking at the failed integration tests triggered by this PR.